homeassistant/components/counter/__init__.py

```diff
@@ -60,21 +76,27 @@
         restore = cfg.get(CONF_RESTORE)
         step = cfg.get(CONF_STEP)
         icon = cfg.get(CONF_ICON)
+        minimum = cfg.get(CONF_MINIMUM)
+        maximum = cfg.get(CONF_MAXIMUM)
 
-        entities.append(Counter(object_id, name, initial, restore, step, icon))
+        entities.append(Counter(object_id, name, initial, minimum, maximum,
+                                restore, step, icon))
 
     if not entities:
         return False
 
     component.async_register_entity_service(
-        SERVICE_INCREMENT, SERVICE_SCHEMA,
+        SERVICE_INCREMENT, SERVICE_SCHEMA_SIMPLE,
         'async_increment')
     component.async_register_entity_service(
-        SERVICE_DECREMENT, SERVICE_SCHEMA,
+        SERVICE_DECREMENT, SERVICE_SCHEMA_SIMPLE,
         'async_decrement')
     component.async_register_entity_service(
-        SERVICE_RESET, SERVICE_SCHEMA,
+        SERVICE_RESET, SERVICE_SCHEMA_SIMPLE,
         'async_reset')
+    component.async_register_entity_service(
+        SERVICE_SETUP, SERVICE_SCHEMA_SETUP,
+        'async_setup')
```

Comment:
Remove that way of setup. That is not like the core works. You can set a config flow and setup an additional counter. Or we discover a new core way. But that is hacky.

Reply:
@pvizeli: This was originally suggested by @dgomez in the first PR (https://github.com/home-assistant/home-assistant/pull/17440#issuecomment-430569178) and @balloob was okay with it then (six months ago). Has our stance on this changed over that time period?

Reply:
I believe there is a misunderstanding, it is not a **HA core setup** call, it is this specific componente (counter) **internal parameters setup** call

Reply:
@balloob @pvizeli are we good with this as is?

Reply:
Let's change the name to avoid confusion. `async_configure`

Reply:
Makes sense, changes pushed up.